src/Rule/HasLength.php

```diff
@@ -36,56 +38,64 @@ final class HasLength implements
     use SkipOnErrorTrait;
     use WhenTrait;
 
+    /**
+     * @param int|null $min Minimum length. null means no minimum length limit. Can't be combined with
+     * {@see $exactly}. See {@see $lessThanMinMessage} for the customized message for a too short string.
+     * @param int|null $max maximum length. null means no maximum length limit. Can't be combined with
+     * {@see $exactly}. See {@see $greaterThanMaxMessage} for the customized message for a too long string.
+     * @param int|null $exactly Exact length. `null` means no strict comparison. Mutually exclusive with
+     * {@see $min} and {@see $max}.
+     * @param string $incorrectInputMessage Error message used when the value is not a string.
+     *
+     * You may use the following placeholders in the message:
+     *
+     * - `{attribute}`: the label of the attribute being validated.
+     * - `{type}`: the type of the attribute being validated.
```

Comment:
```suggestion
     * - `{type}`: the type of the value being validated.
```

Reply:
Because there is a case with single value when attribute is not presented at all.

Reply:
Actual for other occurrences too.